src/sentry/utils/cloudfunctions.py

```diff
@@ -18,51 +18,39 @@
 from sentry.utils import json
 
 WRAPPER_JS = """
+const Sentry = require('@sentry/serverless');
 const userFunc = require('./function.js');
-Object.assign(process.env, require('./env.json'));
-const Sentry = require("@sentry/node");
-const Tracing = require("@sentry/tracing");
-exports.start = async (message, context) => {
-  if (process.env.SENTRY_DSN) {
-    Sentry.init({
-      dsn: process.env.SENTRY_DSN,
-      integrations: [
-        new Sentry.Integrations.Http({ tracing: true }),
-      ],
-      tracesSampleRate: 1.0,
-    });
-  }
-  if (!userFunc) {
-    console.error("Your code needs to export a function. module.export = () => {}");
-    return;
-  }
-  if (typeof userFunc !== 'function') {
-    console.error("Your code needs to export a function. Instead, " + typeof userFunc + " was exported.");
-    return;
-  }
-  const event = JSON.parse(Buffer.from(message.data, 'base64').toString());
+
+if (process.env.SENTRY_DSN) {
+  Sentry.GCPFunction.init({
+    dsn: process.env.SENTRY_DSN,
+    tracesSampleRate: 1.0,
+  });
+}
+exports.start = Sentry.GCPFunction.wrapCloudEventFunction((context, callback) => {
+  const event = JSON.parse(Buffer.from(context.data, 'base64').toString());
   if (process.env.SENTRY_DSN) {
     const transaction = Sentry.startTransaction({
-        op: "func",
-        name: "Sentry Function Transaction",
+      op: 'faas',
+      name: 'Sentry Function Transaction',
     });
     try {
-        await userFunc(event.data, event.type);
+      userFunc(event.data, event.type);
     } catch (e) {
-        Sentry.captureException(e);
+      Sentry.captureException(e);
     } finally {
-        transaction.finish();
+      transaction.finish();
+      Sentry.flush();
```

Comment:
All of this shouldn't be needed, the `wrapCloudEventFunction` func should be handling this logic (Creating a new transaction, automatically catching errors, automatically flushing from the queue): https://github.com/getsentry/sentry-javascript/blob/549370c7709aa96df4608b2b204bf06e23cd67ea/packages/serverless/src/gcpfunction/cloud_events.ts#L24